exporter/exporter.go

```diff
@@ -7,20 +7,43 @@ import (
 	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
+// Exporter holds Exporter methods and attributes.
 type Exporter struct {
 	client *mongo.Client
 }
 
+// Opts holds new exporter options.
 type Opts struct {
 	DSN string
 }
 
+// New connects to the database and returns a new Exporter instance.
 func New(opts *Opts) (*Exporter, error) {
 	if opts == nil {
 		opts = new(Opts)
 	}
 
-	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(opts.DSN))
+	client, err := connect(context.Background(), opts.DSN)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Exporter{
+		client: client,
+	}, nil
+}
+
+// Disconnect from the database.
+func (e *Exporter) Disconnect(ctx context.Context) error {
```

Comment:
When we will call this method?

If we are going to support both "exporter is always connected" and "exporter re-connects on each scrape" modes, then client can't be a field of the shared object.

Reply:
removed